src/libcollections/str.rs

```diff
@@ -96,11 +96,11 @@ impl<S: Str> SliceConcatExt<str, String> for [S] {
         }
 
         // `len` calculation may overflow but push_str will check boundaries
-        let len = s.iter().map(|s| s.as_slice().len()).sum();
+        let len = s.iter().map(|s| s.as_str().len()).sum();
```

Comment:
There are a bunch of warnings about places I could use slice syntax, I figured I'd do this in two passes to make landing it feasible, although this was less churn than I expected.

I also wasn't sure if the syntax was dependant on a snapshot.